File: sol/solution/0301-0400/0303/Solution.scala
class NumArray(_nums: Array[Int]) {
    val prefixSums: Array[Int] = {
        val n = _nums.length
        val sums = new Array[Int](n + 1)
        sums(0) = 0
        var i = 0
        while (i < n) {
            sums(i + 1) = sums(i) + _nums(i)
            i += 1
        }
        sums
    }
    def sumRange(left: Int, right: Int): Int = {
       prefixSums(right + 1) - prefixSums(left) 
    }
}
/**
 * Your NumArray object will be instantiated and called as such:
 * val obj = new NumArray(nums)
 * val param_1 = obj.sumRange(left,right)
 */

File: sol/solution/0301-0400/0303/Solution.swift
class NumArray {
    private var prefixSums: [Int]
    init(_ nums: [Int]) {
        prefixSums = [0]
        for num in nums {
            prefixSums.append(prefixSums.last! + num)
        }
    }
    func sumRange(_ left: Int, _ right: Int) -> Int {
        return prefixSums[right + 1] - prefixSums[left]
    }
}
/**
 * Your NumArray object will be instantiated and called as such:
 * let obj = NumArray(nums)
 * let ret_1: Int = obj.sumRange(left, right)
 */